packages/smart-router/evm/types/bestTrade.ts

```diff
@@ -1,10 +1,13 @@
-import { BestTradeOptions as BaseBestTradeOptions, ChainId } from '@pancakeswap/sdk'
+import { BestTradeOptions as BaseBestTradeOptions, ChainId, Currency, Pair } from '@pancakeswap/sdk'
 import { Provider as IProvider } from '@ethersproject/providers'
 
 export type Provider = ({ chainId }: { chainId?: ChainId }) => IProvider
 
 export interface BestTradeOptions extends BaseBestTradeOptions {
   provider: Provider
+
+  // If not provided, will use the given provider to fetch pairs on chain
+  getAllCommonPairs?: (one: Currency, another: Currency) => Promise<Pair[]>
```

Comment:
should not be async when coming from hooks?

Reply:
Non promise return allowed